### DISCUSSION TITLE
Power to/ from Raspberry Pi?

### DISCUSSION THREAD

**Discussion by easaw** [awawa-dev/HyperSPI#53] [General]
### Does HyperSPI send power to, or receive power from the Raspberry Pi?

Hi,

The reason I am asking is I would like to use a WeMos D1 R32, which seems to send data to my WS2815 without issues.

The WeMos D1 R32 is powered by 12v, and doesn't work with WS2815 when powered via USB. Therefore I will need to power it via the 12v barrel jack. Here is the board:
https://www.instructables.com/How-to-Set-Up-WeMos-TTgo-ESP32-Uno-D1-R32/
or
https://www.hackster.io/NYH-workshop/wemos-r32-with-arduino-startup-guide-7bc841

I assume there is no power sent to the pi(?) as none of the GPIOs are VCC, but I was wondering if the Clock or Data can _receive_ power, which may be too much for the Pi, if the D1 R32 outputs 5v? (or even 12v!)

Thank you.

**Comment by awawa-dev** [comment]
Hi
Measure the output of the 3V3 pin on the board (or even better: write a simple Arduino program and set the LED pin high and measure it) when the supply is 12V and again when powered by USB and you will probably find the answer. The ESP32 will turn to coal when the supply is 5V or more and the protection diode does not cut it off.

**Reply by awawa-dev** [reply]
> The WeMos D1 R32 is powered by 12v, and doesn't work with WS2815 when powered via USB. Therefore I will need to power it via the 12v barrel jack. 

or the issue is the common ground if the esp board is powered up by the same 12v LED power supply.

**Reply by easaw** [reply]
Re: common ground; is common ground affected if using the same 12v supply, but using a 12v to 5v converter?

Is it no longer 'common'?

Also, it seems to be that WLED doesn't work when that board is powered by 5v, I can't visit the page.
Yet 12v works fine...

_But that is obviously a whole other issue, unrelated to HyperHDR..._

Anyway, thank you for your help, I think the WS2815 is 5v data, I doubt that D1 R32 puts out 12v over GPIO...(?)

**Reply by awawa-dev** [reply]
> Re: common ground; is common ground affected if using the same 12v supply, but using a 12v to 5v converter?
Is it no longer 'common'?

The ground should still be common and you can check it with a multimeter. I suspect that the 12/5V is converted on the board to 3.3V for the esp32. Maybe there is a small difference on the 3.3V line depending on the input voltage? But without a multimeter it's just a guess.

**Reply by easaw** [reply]
So, I tried a multimeter on the data line and ground on the strip, and via the ground on the R32, both show about 0.05v when there is a constant stream of data (on both data lines).
(or did I measure that wrong!?)

_Edit: I should mention this is currently only 30 LEDs for testing..._

I wonder if the 12v negates the need for the GPIO to provide much power..?

I'm going to try using the R32 with HyperSPI, I'm just slightly worried about that 12v somehow jumping to the Pi, or there being a common ground issue...

Might put a MOFFSET in there...
![moff](https://duckduckgo.com/i/c9a6d7cb65c3b45d.jpg)

_err... I mean MOSFET, thanks Walter...._

**Reply by awawa-dev** [reply]
> So, I tried a multimeter on the data line and ground on the strip, and via the ground on the R32, both show about 0.05v when there is a constant stream of data (on both data lines).
(or did I measure that wrong!?)

if you don't have an oscilloscope, you won't be able to measure the maximum voltage on the data line - it changes too fast for multimeter: you'll need to write a simple Arduino program that sets some pin high (to constant 3.3V) or measure pin 3v3. Ground should be connected from GPIO to LED strip rather than relying on a complicated power supply scheme (in case of a 12->5v converter)

**Reply by easaw** [reply]
Right, but 3v3 will always be 3v3? Or no?

I'll have put a few lines together in Arduino because I don't have any professional tech like that.

Thanks!

**Reply by awawa-dev** [reply]
It may depends on the regulator and the input voltage how precise it is. Why dont you just measure it?
